func.go

```diff
@@ -163,6 +164,16 @@ func RegisterFunc(fptr interface{}, cfn uintptr) {
 				panic("purego: unsupported kind " + arg.Kind().String())
 			}
 		}
+		if ty.NumOut() > 0 && ty.Out(0).Kind() == reflect.Struct {
+			if runtime.GOOS != "darwin" {
+				panic("purego: struct return values only supported on darwin arm64 & amd64")
+			}
+			outType := ty.Out(0)
+			checkStructFieldsSupported(outType)
+			if outType.Size() > 16 {
+				ints++
```

Comment:
Why don't we check for ints > numOfIntegerRegisters() here?

Reply:
The check is done after this at L177